pkg/reconciler/openshift/openshift.go

```diff
@@ -22,6 +24,12 @@ func reconcilerHook(cr *argoprojv1alpha1.ArgoCD, v interface{}) error {
 				o.Rules = append(o.Rules, policyRulesForClusterConfig()...)
 			}
 		}
+	case *appsv1.Deployment:
+		if o.ObjectMeta.Name == cr.Name+"-redis" {
+			o.Spec.Template.Spec.Containers = []corev1.Container{{
+				Args: append([]string{"redis-server"}, o.Spec.Template.Spec.Containers[0].Args...),
```

Comment:
The argument list of Red Hat Redis looks like this:      
```
      - args:
        - redis-server
        - --save
        - ""
        - --appendonly
        - "no"
        - --protected-mode
        - "no"
```

Reply:
The order does not matter.  I guess you can have `--protected-mode no` in front of `--save`.

Reply:
You can test with `registry.redhat.io/rhel8/redis-5` image.

Reply:
So what we are doing here is inserting `redis-server` arg in the existing set of args defined for stock redis. This makes our arg list look something like this 
```
      - args:
        - redis-server
        - --save
        - ""
        - --appendonly
        - "no"
 ```
Do we need turn off the protected mode explicitly? since we didn't do this for the stock redis !!

Reply:
I think we do.   The stock image runs with protected mode by default.

Reply:
@shubhamagarwal19 @iam-veeramalla @bigkevmcd Can you explain the go mechanism to "compile out"  the openshift specific package by default?

Reply:
@wtam2018 I'm not sure if you're familiar or not with Go build tags, but you can read about them here https://golang.org/pkg/go/build/#hdr-Build_Constraints

The way the extension functions (hooks) work is that they register on import from the init function, and the build tags for the [manager](https://github.com/argoproj-labs/argocd-operator/blob/master/cmd/manager/openshift.go) only trigger the import when the openshift tags are provided.

This will have no impact on the gitops-operator, because it doesn't use the same `main` function, _but_ it can import the same package to register the hooks, and because it's openshift specific, it doesn't need conditional compilation tags, but could easily be configured with them.

Reply:
@wtam2018, i verified the changes using the image `registry.redhat.io/rhscl/redis-5-rhel7:latest` without specifying `--protected-mode` arg. works fine !!

Reply:
Added `protected-mode=no` to the list of args for redhat based redis since stock redis turns the `protected-mode` off by default.